iwyu_preprocessor.cc

```diff
@@ -400,6 +400,11 @@ void IwyuPreprocessorInfo::MaybeProtectInclude(
     protect_reason = "pragma_keep";
     FileInfoFor(includer)->ReportKnownDesiredFile(includee);
 
+  } else if (ShouldKeepIncludeFor(includee)) {
+    // The command line version of pragma keep.
+    protect_reason = "pragma_keep";
```

Comment:
This reason is used for logging, so maybe better not to attribute this to "pragma: keep". Let's say `--keep` instead.